WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift

```diff
@@ -479,50 +470,52 @@ import Foundation
             let button = buttons.removeLast() as UIButton
             configureCommentActionButton(button)
         }
-
-        // Show visit
-        if UIDevice.isPad() {
-            let button = buttons.removeLast() as UIButton
-            configureVisitActionButton(button)
-        } else {
-            configureVisitActionButton(actionButtonFlushLeft)
-        }
     }
 
     private func resetActionButtons() {
-        resetActionButton(actionButtonCenter)
-        resetActionButton(actionButtonFlushLeft)
         resetActionButton(actionButtonLeft)
         resetActionButton(actionButtonRight)
     }
 
     private func resetActionButton(button:UIButton) {
         button.setTitle(nil, forState: .Normal)
+        button.setTitle(nil, forState: .Highlighted)
         button.setImage(nil, forState: .Normal)
         button.setImage(nil, forState: .Highlighted)
         button.selected = false
         button.hidden = true
     }
 
-    private func configureActionButton(button: UIButton, title: String?, image: UIImage?, highlightedImage: UIImage?) {
+    private func configureActionButton(button: UIButton, title: String?, image: UIImage?, highlightedImage: UIImage?, selected:Bool) {
         button.setTitle(title, forState: .Normal)
+        button.setTitle(title, forState: .Highlighted)
         button.setImage(image, forState: .Normal)
         button.setImage(highlightedImage, forState: .Highlighted)
-        button.selected = false
+        button.selected = selected
         button.hidden = false
     }
 
     private func configureLikeActionButton(button: UIButton) {
         button.tag = CardAction.Like.rawValue
         let likeStr = NSLocalizedString("Like", comment: "Text for the 'like' button. Tapping marks a post in the reader as 'liked'.")
-        let likedStr = NSLocalizedString("Liked", comment: "Text for the 'like' button. Tapping removes the 'liked' status from a post.")
-        let title = contentProvider!.isLiked() ? likedStr : likeStr
+        let likesStr = NSLocalizedString("Likes", comment: "Text for the 'like' button. Tapping removes the 'liked' status from a post.")
+
+        let count = contentProvider!.likeCount().integerValue
+
+        var title: String?
```

Comment:
I'd suggest moving the logic to define the title into a method of it's own so that the only thing that'll remain in this method is:

```swift
let title = self. likeButtonTitleForNumberOfLikes(count)
```

The new method would be something along the lines of:

```swift
private func likeButtonTitleForNumberOfLikes(count: Int) -> String {
    let likeStr = NSLocalizedString("Like", comment: "Text for the 'like' button. Tapping marks a post in the reader as 'liked'.")
    let likesStr = NSLocalizedString("Likes", comment: "Text for the 'like' button. Tapping removes the 'liked' status from a post.")

    if count == 0 {
        return likeStr
    } else if count == 1 {
        return "\(count) \(likeStr)"
    } else {
        return "\(count) \(likesStr)"
    }
}
```

I didn't compile the code above so it may not be the exact same result - it's just a rough approximation to better convey mi thoughts.